clojure-mode.el

```diff
@@ -1534,6 +1546,197 @@ This will skip over sexps that don't represent objects, so that ^hints and
           (backward-sexp 1))
         (setq n (1- n))))))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Refactoring support
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(defcustom clojure-thread-all-but-last nil
+  "When true `cljr-thread-first-all' and `cljr-thread-last-all' don't thread the last expression."
+  :package-version '(clojure-mode . "5.3.0")
+  :safe #'booleanp
+  :type 'boolean)
+
+(defun clojure--unwind-last ()
+  (forward-sexp)
+  (save-excursion
+    (let ((contents (clojure-delete-and-extract-sexp)))
+      (when (looking-at " *\n")
+        (join-line -1))
+      (clojure--ensure-parens-around-function-names)
+      (let* ((sexp-beg-line (line-number-at-pos))
+             (sexp-end-line (progn (forward-sexp)
+                                   (line-number-at-pos)))
+             (multiline-sexp-p (not (= sexp-beg-line sexp-end-line))))
+        (down-list -1)
+        (when multiline-sexp-p
+          (newline))
+        (insert contents)
+        (when multiline-sexp-p
+          (clojure-indent-line)))))
+  (forward-char))
+
+(defun clojure--wrap-around ()
+  (insert (format "(%s)" (clojure-delete-and-extract-sexp))))
+
+(defun clojure--ensure-parens-around-function-names ()
+  (unless (looking-at "[\n\r\t ]?(")
+    (skip-syntax-forward " ")
```

Comment:
Actually, calling `(clojure--looking-at-non-logical-sexp)` would skip both whitespace *and* comments, so it might be a better replacement for `(skip-syntax-forward "\n\r\t ")`.